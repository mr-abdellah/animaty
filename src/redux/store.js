import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import animeReducer from './cards';

const store = configureStore({
  reducer: {
    animes: animeReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
