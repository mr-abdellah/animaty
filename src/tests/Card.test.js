import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CardItem from '../components/Card';

it('renders properly', () => {
  const tree = renderer
    .create(
      <Router>
        <Provider store={store}>
          <CardItem />
        </Provider>
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
