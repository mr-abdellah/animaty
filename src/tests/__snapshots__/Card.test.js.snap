// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Card renders properly 1`] = `
<div
  className="card-item"
>
  <img
    alt=""
    src={null}
  />
  <h4
    id="card-title"
  />
  <p>
    Episodes : null
  </p>
  <a
    href="/anime/"
    id="card-btn"
    onClick={[Function]}
  >
    More Details
  </a>
</div>
`;
